PayoutCurve.md

```diff
@@ -125,36 +186,74 @@ Please note that while the following may seem complex, it should boil down to ve
 Stack Overflow from which they can be copied so that only minor aesthetic modifications are required.
 
-Given a potential `event_outcome` compute the `outcome_payout` as follows:
+Given a potential `event_outcome` compute the `outcome_payout` as follows (where points is a list including the `endpoint`s):
 
-1. Binary search the endpoints by `event_outcome`
-   * If found, return `outcome_payout`
-   * Else let `points` be all of the interpolation points between (inclusive)
-     the previous and next endpoint
-2. Let `lagrange_line(i, j) = (event_outcome - points(j).event_outcome)/(points(i).event_outcome - points(j).event_outcome)`
-3. Let `lagrange(i) := PROD(j = 0, j < points.length && j != i, lagrange_line(i, j))`
-4. Return `SUM(i = 0, i < points.length, points(i).outcome_payout * lagrange(i))`
+1. Let `lagrange_line(i, j) = (event_outcome - points(j).event_outcome)/(points(i).event_outcome - points(j).event_outcome)`
+2. Let `lagrange(i) := PROD(j = 0, j < points.length && j != i, lagrange_line(i, j))`
+3. Return `SUM(i = 0, i < points.length, points(i).outcome_payout * lagrange(i))`
 
-#### Reference Implementations
+### Hyperbola Curve Piece
 
-* [bitcoin-s](https://github.com/bitcoin-s/bitcoin-s/blob/adaptor-dlc/core/src/main/scala/org/bitcoins/core/protocol/dlc/DLCPayoutCurve.scala#L365)
+The goal of this specification is to enable general hyperbola shapes efficiently and compactly while also ensuring that simpler and more common payout curves (such as `constant/outcome`) do not become complex while conforming to the generalized structure.
 
-### Optimized Evaluation During CET Calculation
+This is accomplished by using a total of 7 parameters which can (almost) uniquely express every affine transformation of the curve `1/x`.
+Specifically we have `(f_1, f_2) + ((a, b), (c, d))*(x, 1/x)` which is some translation by the point `(f_1, f_2)` added to the curve `(x, 1/x)` transformed by the matrix `((a, b), (c, d))` where `a*d =/= b*c`.
+The last parameter is a boolean used to specify which curve piece to use when there is ambiguity.
 
-There are many optimizations to this piecewise interpolation function that can be made
-when repeatedly and sequentially evaluating an interpolation as is done during CET calculation.
+This scheme keeps simple curves simple as to represent any curve of the form `constant/x + constant'` we simply
+set `f_1 = b = c = 0, a = 1, d = constant, f_2 = constant'`.
 
-* The binary search in step 1 can be avoided when computing for sequential inputs.
+#### Hyperbola Serialization
 
-* The value ` points(i).outcome_payout / PROD(j = 0, j < points.length && j != i, points(i).event_outcome - points(j).event_outcome)`
-  can be cached for each `i` in a polynomial piece, call this `coef_i`.
-  For a given `event_outcome` let `all_prod = PROD(i = 0, i < points.length, event_outcome - points(i).event_outcome)`.
+1. type: ??? (`hyperbola_payout_curve_piece`)
```

Comment:
Should there be a version number for `polynomial_payout_curve_piece` and `hyperbola_payout_curve_piece`?